brayns/engine/LightManager.h

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) 2019, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+#include <brayns/api.h>
+#include <brayns/common/BaseObject.h>
+#include <brayns/common/types.h>
+
+namespace brayns
+{
+class LightManager : public BaseObject
+{
+public:
+    /**
+     * @brief addLight Attaches a light source to the scene
+     * @param light Object representing the light source
+     */
+    void addLight(LightPtr light);
+
+    /**
+     * @brief removeLight Removes a light source from the scene for a given id
+     * @param id ID of Light source to be removed
+     */
+    void removeLight(const size_t id);
+
+    /**
+     * @brief removeLight Removes a light source from the scene for a given
+     * light
+     * @param light Light source to be removed
+     */
+    void removeLight(LightPtr light);
+
+    /**
+     * @brief getLight Gets a light source from the scene for a given id.
+     * Note: If changing the light then call markModified to propagate the
+     * changes.
+     * @param id Id of the light
+     * @return Pointer to the light, nullptr if not found
+     */
+    LightPtr getLight(const size_t id);
+
+    /**
+     * @brief getLights Gets the lights
+     * @return Immutable list of all light sources
+     */
+    const std::vector<LightPtr>& getLights() const;
+
+    /**
+     * @brief clearLights Removes all light sources
+     */
+    void clearLights();
+
+    /**
+     * @brief addDirectionalLight Creates a light that shines from a specific
+     * direction not from a specific position. This light will behave as though
+     * it is infinitely far away and the rays produced from it are all parallel
+     * @param direction Light source direction
+     * @param color Light source RGB color
+     * @param intensity Amount of light emitted
+     */
+    LightPtr addDirectionalLight(const Vector3f& direction,
```

Comment:
This `addXYZLight` should be either free functions or static member functions named `createXYZLight`. I would even move them to the Light class. For adding the light, client code would only use `LightManager::addLight`